File: ACM-ECPC-2017/Solutions/K-Katryoshka.cpp
#include <bits/stdc++.h>
using namespace std;

int t;

int main() {
	//freopen("katryoshka.in", "rt", stdin);
	cin >> t;
	for (int tt = 1; tt <= t; tt++) {
		long long ans = 0;
		int eye, mo, bod;
		cin >> eye >> mo >> bod;
		int mn = min(min(eye, mo), bod);
		eye -= mn;
		mo -= mn;
		bod -= mn;
		ans += mn;
		ans += min(eye / 2, bod);
		cout << "Case " << tt << ": " << ans << '\n';
	}
	return 0;
}

File: ACM-ECPC-2017/Solutions/K-Katryoshka.py
'''
Author: @vinuthegr8
Submission Link: https://codeforces.com/gym/101856/submission/58209101
'''

import sys
import math
input = sys.stdin.readline
 
f = open("katryoshka.in",'r')
flag=True
case = 1
for i in f:
	if flag:
		flag=False
		t=int(i)
	else:
		n,m,k=map(int,i.split())
		b=min(n,m,k)
		n-=b
		m-=b
		k-=b
		b+=min(n//2,k)
		print("Case " + str(case) + ": " + str(b))
		case += 1
